src/mo_frontend/typing.ml

```diff
@@ -1115,7 +1117,7 @@ and infer_exp' f env exp : T.typ =
   if not env.pre then begin
     assert (T.normalize t' <> T.Pre);
     let e = A.infer_effect_exp exp in
```

Comment:
I'm surprised omitting the normalization doesn't break the type checker. I'm curious why you do this. Is it avoid losing type names to better map to Viper's nominal type system?

Reply:
Maybe it is just a leftover from early development stages?